controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -350,6 +373,32 @@ func (r *ROSAControlPlaneReconciler) reconcileDelete(ctx context.Context, rosaSc
 	return ctrl.Result{RequeueAfter: time.Second * 60}, nil
 }
 
+// isMachinePoolsDeleted check if the controlplane related machinePools are deleted.
+func (r *ROSAControlPlaneReconciler) isMachinePoolsDeleted(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (bool, error) {
+	machinePools, err := rosaScope.GetMachinePools(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	var errs []error
+	for id, mp := range machinePools {
+		if !mp.DeletionTimestamp.IsZero() {
+			continue
+		}
+		err = rosaScope.Client.Delete(context.TODO(), &machinePools[id])
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
```

Comment:
```suggestion
		if err := rosaScope.Client.Delete(context.TODO(), &machinePools[id]); err != nil {
			errs = append(errs, err)
		}
```
no need for `continue`